src/components/Grunnlag/GrunnlagInntekt/GrunnlagInntekt.tsx

```diff
@@ -0,0 +1,273 @@
+        })
+      )
+      logger('button klikk', {
+        tekst: 'endrer pensjonsgivende inntekt',
+      })
+      /* c8 ignore next 3 */
+      if (inntektModalRef.current?.open) {
+        inntektModalRef.current?.close()
+      }
+    }
+  }
+
+  const onCancel = (): void => {
+    setValidationError('')
+    inntektModalRef.current?.close()
+  }
+
+  const isInntektGreaterThanZero =
+    aarligInntektFoerUttakFraBrukerInput !== null ||
+    (aarligInntektFoerUttakFraBrukerInput === null &&
+      aarligInntektFoerUttakFraSkatt &&
+      aarligInntektFoerUttakFraSkatt.beloep > 0)
+
+  return (
+    <>
+      <Modal
+        ref={infoModalRef}
+        className={styles.modal}
+        header={{
+          heading: intl.formatMessage({
+            id: 'grunnlag.inntekt.infomodal.title',
+          }),
+        }}
+      >
+        <Modal.Body>
+          <Label as="h2">
+            <FormattedMessage id="grunnlag.inntekt.infomodal.subtitle" />
+          </Label>
+          <ul>
+            <li>
+              <FormattedMessage id="grunnlag.inntekt.infomodal.list_item1" />
+            </li>
+            <li>
+              <FormattedMessage id="grunnlag.inntekt.infomodal.list_item2" />
+            </li>
+            <li>
+              <FormattedMessage id="grunnlag.inntekt.infomodal.list_item3" />
+            </li>
+            <li>
+              <FormattedMessage id="grunnlag.inntekt.infomodal.list_item4" />
+            </li>
+            <li>
+              <FormattedMessage id="grunnlag.inntekt.infomodal.list_item5" />
+            </li>
+            <li>
+              <FormattedMessage id="grunnlag.inntekt.infomodal.list_item6" />
+            </li>
+            <li>
+              <FormattedMessage id="grunnlag.inntekt.infomodal.list_item7" />
+            </li>
+            <li>
+              <FormattedMessage id="grunnlag.inntekt.infomodal.list_item8" />
+            </li>
+            <li>
+              <FormattedMessage id="grunnlag.inntekt.infomodal.list_item9" />
+            </li>
+          </ul>
+          <BodyLong>
+            <FormattedMessage
+              id="grunnlag.inntekt.infomodal.ingress"
+              values={{
+                ...formatMessageValues,
+              }}
+            />
+          </BodyLong>
+        </Modal.Body>
+        <Modal.Footer>
+          <Button type="button" onClick={() => infoModalRef.current?.close()}>
+            {intl.formatMessage({ id: 'grunnlag.inntekt.infomodal.lukk' })}
+          </Button>
+        </Modal.Footer>
+      </Modal>
+      <Modal
+        ref={inntektModalRef}
+        className={styles.modal}
+        header={{
+          heading: intl.formatMessage({
+            id: 'grunnlag.inntekt.inntektmodal.title',
+          }),
+        }}
+      >
+        <Modal.Body>
+          <form id="oppdatere-inntekt" method="dialog" onSubmit={onSubmit}>
+            <TextField
+              data-testid="inntekt-textfield"
+              type="number"
```

Comment:
Aksel råder i mot `type="number"`. Kunne kanskje brukt `type="text" inputMode="numeric"` og legge validering dersom man skrive tekst?
https://aksel.nav.no/komponenter/core/textfield

Da slipper man unna stepperene til høyre og den innebygde stylingen for feilmeldinger
![image](https://github.com/navikt/pensjonskalkulator-frontend/assets/2196936/3323dfda-96c1-4605-ae8b-ff466ee6c8d7)

Reply:
Veldig bra innspill. Har tatt en runde på det med Cecilie og gikk bort fra type="number" og forbedret validering med flere sjekk og bedre feilmeldinger.